### PR TITLE
Closes #57

### LINKED ISSUE #57: `LogisticLogNormal` class clean up
To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #57

### CODE REVIEW

**Comment by wwojciech** [comment]
Hi @danielinteractive I think that the description of `.LogisticLogNormal` could probably be improved a bit. It is quite difficult for me to do it now, as I do not have yet a full visibility on the usage context of this function.

File: R/Model-class.R [hunk 1]
```diff
@@ -127,144 +127,130 @@ NULL
...
+#' @details The covariate is the natural logarithm of the dose \eqn{x} divided
+#'   by the reference dose \eqn{x^{*}}:
+#'   \deqn{logit[p(x)] = \alpha + \beta \cdot \log(x/x^{*})},
+#'   where \eqn{p(x)} is the probability of observing a DLT for a given dose
+#'   \eqn{x}.
+#'   The prior is: \deqn{(\alpha, \log(\beta)) \sim Normal(\mu, \Sigma)}.
+#'   The slots of this class contain the mean vector and the covariance matrix
+#'   of the bivariate normal distribution, as well as the reference dose.
+#'
+#' @note Te parametrization inside the class uses `alpha0` and `alpha1`.
```
Comment (danielinteractive) [inline R/Model-class.R:148]:
```suggestion
#' @note The parametrization inside the class uses `alpha0` and `alpha1`.
```

File: R/Model-class.R [hunk 2]
```diff
@@ -127,144 +127,130 @@ NULL
...
+#'   `alpha0` is identical to the intercept \eqn{\alpha} above and is the
+#'   log-odds for a DLT at the reference dose `x`*. Therefore, the prior mean
```
Comment (danielinteractive) [inline R/Model-class.R:150]:
```suggestion
#'   log-odds for a DLT at the reference dose `x*`. Therefore, the prior mean
```
and similar below?

File: R/Model-class.R [hunk 3]
```diff
@@ -127,144 +127,130 @@ NULL
...
+#'   for `alpha0` is the expected log-odds at the reference dose `x`* before
+#'   observing any data. Note that the expected odds is not just the exp of the
+#'   prior mean of `alpha0`, because the non-linearity of the exp
+#'   transformation. The log-normal distribution on Wikipedia gives the formula
+#'   for computing the prior mean of `exp(alpha0)`, where `alpha0` is the
+#'   `log(alpha)` in the Neuenschwander et al. (2008) paper. The `alpha1` is
+#'   identical to \eqn{\beta} above and equals to `beta` in the Neuenschwander
+#'   et al paper. The `exp(alpha1)` gives the odds-ratio for DLT between two
+#'   doses that differ by the factor `exp(1)` ~ 2.7. Paramter `alpha1` has a
```
Comment (danielinteractive) [inline R/Model-class.R:159]:
```suggestion
#'   doses that differ by the factor `exp(1)` ~ 2.7. Parameter `alpha1` has a
```

Comment (danielinteractive) [reply R/Model-class.R:159]: can you double check the wordlist file that it does not contain wrong spelling terms? thx

File: R/helpers.R [hunk 4]
```diff
@@ -665,3 +665,36 @@ h_check_fun_formals <- function(fun, mandatory = NULL, allowed = NULL) {
 
   mandatory_check && allowed_check
 }
+
+#' Test matrix for positive definiteness
```
Comment (danielinteractive) [inline R/helpers.R:669]:
```suggestion
#' Testing Matrix for Positive Definiteness
```
(title style)

File: R/helpers.R [hunk 5]
```diff
@@ -665,3 +665,36 @@ h_check_fun_formals <- function(fun, mandatory = NULL, allowed = NULL) {
...
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' This helper function checks whether a given numeric matrix `x`
+#' is positive-define.
```
Comment (danielinteractive) [inline R/helpers.R:674]:
```suggestion
#' is positive-definite.
```

**Review by danielinteractive — APPROVED** [review]
thx Wojciech, only minor comments inside, good work

File: R/Model-class.R (hunk 3 above)
Comment (wwojciech) [reply R/Model-class.R:159]: Indeed, Tx @danielinteractive , The`WORDLIST` contains wrong `paramter`. I will correct it now!

**Pull request merged**